mixed/flink/flink-common/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory

```diff
@@ -17,4 +17,5 @@
 #
 
 com.netease.arctic.flink.catalog.factories.ArcticCatalogFactory
+com.netease.arctic.flink.catalog.factories.FlinkCatalogFactory
 com.netease.arctic.flink.table.DynamicTableFactory
```

Comment:
After some rethinking, I've removed the factory from services, since it mainly exposes the ability to support create table statment, while the patch mainly provides a unified catalog for creating catalog statements, and I can reopen an issue to support the ability to If necessary. For example:
```
CREATE TABLE AMORO_TBL (...) 
WITH (
  'type'='amoro',
  'metastore.url'='',
  'database'='',
  'table'=''
)
```

And aslo, this patch support create table in the unified catalog. for example:
```
CREATE TABLE UNIFIED_CATALOG.DB.AMORO_TBL (...) 
WITH (
  'table.format'='MIXED_HIVE'
)
```